pyrato/parameters.py

```diff
@@ -102,3 +103,110 @@ def reverberation_time_linear_regression(
         return reverberation_times, intercepts
     else:
         return reverberation_times
+
+
+
+
+
+
+def clarity(energy_decay_curve, early_time_limit=80):
+    r"""
+    Calculate the clarity from the energy decay curve (EDC) of
+    a room impulse response.
+
+    The clarity parameter (C50 or C80) is defined as the ratio of early-to-late
+    arriving energy in an impulse response and is a measure for how clearly
+    speech or music can be perceived in a room. The early-to-late boundary is
+    typically set at 50 ms (C50) or 80 ms (C80) [#iso]_.
+
+    Clarity is calculated as:
+
+    .. math::
+
+        C_{t_e} = 10 \log_{10} \frac{
+            \displaystyle \int_0^{t_e} p^2(t) \, dt
+        }{
+            \displaystyle \int_{t_e}^{\infty} p^2(t) \, dt
+        }
+
+    where :math:`t_e` is the early time limit and :math:`p(t)` is the pressure
+    of a room impulse response. The clarity can also be efficiently computed
+    from the EDC directly by:
+
+    .. math::
+
+        C_{t_e} = 10 \log_{10} \left( \frac{
+            \text{EDC}(t_0)
+        }{
+            \text{EDC}(t_e)
+        } - 1 \right)
+
+    where :math:`t_0` is the EDC start time.
+
+    Parameters
+    ----------
+    energy_decay_curve : pyfar.TimeData
+        Energy decay curve (EDC) of the room impulse response
+        (time-domain signal). The EDC must start at time zero.
+    early_time_limit : float, optional
+        Early time limit (:math:`t_e`) in milliseconds. Defaults to 80 (C80).
+        Typical values are 50 ms (C50) or 80 ms (C80) [#iso]_.
+
+    Returns
+    -------
+    clarity : ndarray of float
+        Clarity index (early-to-late energy ratio) in decibels,
+        shaped according to the channel shape of the input EDC.
+
+    References
+    ----------
+    .. [#iso] ISO 3382, Acoustics — Measurement of the reverberation time of
+        rooms with reference to other acoustical parameters.
+
+    Examples
+    --------
+    Estimate the clarity from a real room impulse response filtered in
+    octave bands:
+
+    >>> import numpy as np
+    >>> import pyfar as pf
+    >>> import pyrato as ra
+    >>> rir = pf.signals.files.room_impulse_response(sampling_rate=44100)
+    >>> rir = pf.dsp.filter.fractional_octave_bands(rir)
+    >>> edc = ra.edc.energy_decay_curve_lundeby(rir)
+    >>> C80 = clarity(edc, early_time_limit=80)
+    """
+
+    # Check input type
+    if not isinstance(energy_decay_curve, pf.TimeData):
+        raise TypeError("Input must be a pyfar.TimeData object.")
+    if not isinstance(early_time_limit, (int, float)):
+        raise TypeError('early_time_limit must be a number.')
+
+    # Validate time range
+    if (early_time_limit > energy_decay_curve.signal_length * 1000) or (early_time_limit <= 0): # noqa: E501
```

Comment:
We try to avoid lines with over lengths whereever possible:
```suggestion
    if (early_time_limit > energy_decay_curve.signal_length * 1000) or (
            early_time_limit <= 0):
```